akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1361,6 +1361,7 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
      * Push to this output port.
      */
     def push(elem: T): Unit = {
+      require(isAvailable, s"Cannot push port ($this) twice, or before it being pulled")
```

Comment:
I think we should have an `if (isAvailable) { throw ... }` here instead of allocating a closure/lambda (second parameter) for every element that is pushed.

Reply:
Ok, we can do that in all places in the GraphStage then to be consistent -- I'll apply that and merge

Reply:
yes, that is important, should not be merged before that is fixed